### PR TITLE
COD-example update to work with HBase 2.4

### PR DESCRIPTION
**Pull request by abhradeepkundu**

### CODE REVIEW

**Review by joshelser — CHANGES_REQUESTED** [review]
Thanks for looking into this Abhradeep! A few suggestions for changes and one open-ended question about the hbase-shaded-client classpath (I haven't gone looking for any docs on HBase's upstream jira as to why otel jars aren't included, so there may be an obvious answer we just have to look for :))

File: phoenix-read-write/README.md [hunk 1]
```diff
@@ -22,65 +22,61 @@ done
...
   ...
 </project>
 ```
 
 # Build the project
 
+While building you can specify profile with -P argument
 ```
-$ mvn package
+$ mvn clean package -P HBASE-2.4
```
[the remark this answers is not in the document]
Comment (abhradeepkundu) [reply phoenix-read-write/README.md:85]: I have made the change accordingly

File: phoenix-read-write/pom.xml [hunk 2]
```diff
@@ -7,20 +7,52 @@
      <maven.compiler.source>1.8</maven.compiler.source>
      <maven.compiler.target>1.8</maven.compiler.target>
    </properties>
+  <profiles>
+    <profile>
+      <id>HBASE-2.2-7.2.8</id>
+      <properties>
+        <pch.artifactid.version>phoenix-client</pch.artifactid.version>
```
[the remark this answers is not in the document]
Comment (abhradeepkundu) [reply phoenix-read-write/pom.xml]: Done

**Comment by abhradeepkundu** [comment]
> Thanks for looking into this Abhradeep! A few suggestions for changes and one open-ended question about the hbase-shaded-client classpath (I haven't gone looking for any docs on HBase's upstream jira as to why otel jars aren't included, so there may be an obvious answer we just have to look for :))

I have done the necessary changes. For Otel jar I still not found anything conclusive yet, still looking.

**Comment by abhradeepkundu** [comment]
Sorry I have mistakenly closed the request earlier, opening it now

**Comment by petersomogyi** [comment]
The `opentelemetry-api` and `opentelemetry-context` are transitive dependencies of hbase-shaded-client so there is no need to include them as dependencies. It needs to be added to `includeArtifactIds` only. Similarly, `htrace` is not needed anymore as `opentelemetry` replaced it. Cloudera Runtime versions before 7.2.14 still require htrace instead of opentelemetry.

```
[INFO] +- org.apache.hbase:hbase-shaded-client:jar:2.4.6.7.2.15.0-147:compile
[INFO] |  +- io.opentelemetry:opentelemetry-api:jar:0.12.0:compile
[INFO] |  +- io.opentelemetry:opentelemetry-context:jar:0.12.0:compile
[INFO] |  \- org.apache.yetus:audience-annotations:jar:0.5.0:compile
```

**Review by petersomogyi — APPROVED** [review]

**Pull request merged**